### PR TITLE
TIP-1171: Association

### PR DESCRIPTION
**Pull request by pierallard**
Cancelled as we have to put a method `addAssociation` in the product.

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php [hunk 1]
```diff
@@ -604,6 +606,8 @@ public function addAssociation(AssociationInterface $association): EntityWithAss
 
             $this->associations->add($association);
             $association->setOwner($this);
+
+            $this->events[] = new AssociationAddedToProduct($this->identifier, $association->getId());
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php]: Maybe check if it was not already there

Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php]: Do we add an association? or do associate a product to another one?

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AssociationAddedToProduct.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        $this->productIdentifier = $productIdentifier;
+        $this->associationId = $associationId;
+    }
+
+    public function productIdentifier(): string
+    {
+        return $this->productIdentifier;
+    }
+
+    public function associationId(): string
```
Comment (arnolanglade) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AssociationAddedToProduct.php]: maybe `associationIdentifier` to be concistent with the other method

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AssociationRemovedFromProduct.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        $this->productIdentifier = $productIdentifier;
+        $this->associationId = $associationId;
+    }
+
+    public function productIdentifier(): string
+    {
+        return $this->productIdentifier;
+    }
+
+    public function associationId(): string
```
Comment (arnolanglade) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AssociationRemovedFromProduct.php]: idem

**Pull request closed without merging**